src/ui/public/state_management/state_storage/state_hash.js

```diff
@@ -0,0 +1,29 @@
+import { Sha256 } from 'ui/crypto';
+
+// This prefix is used to identify hash strings that have been encoded in the URL.
+const HASH_PREFIX = 'h@';
+
+export function createStateHash(json, existingJsonProvider) {
+  if (typeof json !== 'string') {
+    throw new Error('createHash only accepts strings (JSON).');
+  }
```

Comment:
I'd remove the guard-clause. This isn't public API. If we're triggering this error, we're using our own code the wrong way.